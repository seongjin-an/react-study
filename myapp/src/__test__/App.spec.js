import React from "react";
import {cleanup, render, waitFor} from '@testing-library/react'
import MyApp from "../components/basic/JestTestApp";

describe('/////JestTestApp/////', () => {

  afterEach(cleanup)

  function expectListItems(count){
    return waitFor(()=>{
      const lis = document.querySelectorAll('li')
      expect(lis.length).toBe(count)
    })
  }
  test('should render', () => {
    const {getByText} = render(<MyApp age={28}/>)
    // expect(getByText('hello halo'))
    const titleElement = getByText(/hello/i);
    expect(titleElement).not.toBeNull()
    expect(getByText(/28/i)).not.toBeNull()
  })

  test('should match snapshot', () => {
    const {container} = render(<MyApp age={28}/>)
    expect(container).toMatchSnapshot()
  })

  test('show ul children counts', async () => {
    const {getByText} = render(<MyApp age={28}/>)
    await expectListItems(2)
  })
})