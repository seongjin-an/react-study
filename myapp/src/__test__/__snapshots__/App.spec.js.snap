// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/////JestTestApp///// should match snapshot 1`] = `
<div>
  <div
    class="myApp"
  >
    hello halo 
    28
    <ul>
      <li>
        <label>
          이름: 
          <input />
        </label>
      </li>
      <li>
        <label>
          나이: 
          <input />
        </label>
      </li>
    </ul>
  </div>
</div>
`;
